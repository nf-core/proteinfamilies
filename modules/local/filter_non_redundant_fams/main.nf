process FILTER_NON_REDUNDANT_FAMS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/31/313e1c18a344323886cf97a151ab66d81c1a146fb129558cb9382b69a72d5532/data' :
        'community.wave.seqera.io/library/python:b1b4b1f458c605bb' }"

    input:
    tuple val(meta) , path(files, stageAs: "input_folder/*")
    tuple val(meta2), path(redundant_ids)

    output:
    tuple val(meta), path("*.${extension}"), emit: filtered
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    extension = files[0].extension
    """
    filter_non_redundant_fams.py \\
        --input_folder input_folder  \\
        --redundant_ids ${redundant_ids}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    extension = files[0].extension
    """
    touch test.${extension}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}
