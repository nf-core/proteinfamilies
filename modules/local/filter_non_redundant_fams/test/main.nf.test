nextflow_process {

    name "Test Process FILTER_NON_REDUNDANT_FAMS"
    script "../main.nf"
    process "FILTER_NON_REDUNDANT_FAMS"

    test("proteinfamilies - nonredundant") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'aln' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.aln', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.aln', checkIfExists: true)
                    ]
                ])
                input[1] = Channel.of([
                    [ id: 'redundant_ids' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/redundant_fam_ids.txt', checkIfExists: true),
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.filtered,
                    process.out.versions.collect { path(it).yaml }
                    ).match() }
            )
        }
    }

    test("proteinfamilies - nonredundant - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'aln' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.aln', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.aln', checkIfExists: true)
                    ]
                ])
                input[1] = Channel.of([
                    [ id: 'redundant_ids' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/redundant_fam_ids.txt', checkIfExists: true),
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
