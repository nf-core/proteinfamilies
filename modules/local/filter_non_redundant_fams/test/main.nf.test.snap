{
    "proteinfamilies - nonredundant - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "aln"
                        },
                        "test.aln:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,36997366ce1de5272a1208b9c3cad892"
                ],
                "filtered": [
                    [
                        {
                            "id": "aln"
                        },
                        "test.aln:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,36997366ce1de5272a1208b9c3cad892"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-13T15:17:06.261303697"
    },
    "proteinfamilies - nonredundant": {
        "content": [
            [
                [
                    {
                        "id": "aln"
                    },
                    "mgnifams_test_2.aln:md5,62ba4732047ba75151f735c2399973a4"
                ]
            ],
            [
                {
                    "FILTER_NON_REDUNDANT_FAMS": {
                        "python": "3.13.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-13T15:16:53.342516494"
    }
}